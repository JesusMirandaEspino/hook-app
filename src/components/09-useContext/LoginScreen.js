import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    const persona = {
        id: 123458,
        name: 'Jesus'
    }

    return (
        <div>
            <h1> LoginScreen </h1>
            <hr/>

            <button className=" btn btn-primary"  onClick={ () => { setUser( persona ) } }  > Login </button>

        </div>
    )
}
