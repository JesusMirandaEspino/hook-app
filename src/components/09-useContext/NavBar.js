import React from 'react'
import {  Link } from 'react-router-dom';

export const NavBar = () => {
    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
            <Link className="navbar-brand" to="/">Use Context</Link>
            <div clasNames="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item"> <Link className="nav-link active" aria-current="page" to="./">Home</Link> </li>
                    <li className="nav-item"> <Link className="nav-link active" aria-current="page" to="./about">About</Link> </li>
                    <li className="nav-item"> <Link className="nav-link" to="./login">Login</Link> </li>
                </ul>
            </div>
        </div>
        </nav>


    )
}
