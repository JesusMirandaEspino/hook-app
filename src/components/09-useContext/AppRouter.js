import React from 'react';
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>

            <switch>
                <Route exact path="/"  component={ HomeScreen } />
                <Route exact path="/about"  component={ AboutScreen } />
                <Route exact path="/login"  component={ LoginScreen } />
                

            </switch>

            </div>
        </Router>

    )
}
