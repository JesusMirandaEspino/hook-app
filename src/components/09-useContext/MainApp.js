import React from 'react'
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';


export const MainApp = () => {

    const user = {
        id:     12365879,
        name:   'Jesus',
        email:  'jesus@email.com'
    }


    return (
            <UserContext.Provider value={ user }  >

                <AppRouter />

            </UserContext.Provider>
    )
}
