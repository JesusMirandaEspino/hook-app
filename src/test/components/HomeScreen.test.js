
import React from 'react';
import '@testing-library/jest-dom';
import {  mount } from 'enzyme';

import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Pruebas con <HomeScreen />', () => {

    const user = {
        name: 'Jesus',
        email: 'jesus@gmail.com'
    }

    const wrapper = mount(  

        <UserContext.Provider value={{ user }}>
        
            <HomeScreen /> 

         </UserContext.Provider>
         
         
         );

    test( 'Debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

});