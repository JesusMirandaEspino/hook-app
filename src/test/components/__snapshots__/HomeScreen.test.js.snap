// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <HomeScreen /> Debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
     HomeScreen 
  </h1>
  <hr />
  <pre>
    {
   "name": "Jesus",
   "email": "jesus@gmail.com"
}
  </pre>
</div>
`;
